localstack/services/sns/provider.py

```diff
@@ -132,7 +132,8 @@
 
 GCM_URL = "https://fcm.googleapis.com/fcm/send"
 
-MSG_ATTR_NAME_REGEX = r"^(?!\.)(?!.*\.$)(?!.*\.\.)[a-zA-Z0-9_\-.]+$"
+MSG_ATTR_NAME_REGEX = re.compile(r"^(?!\.)(?!.*\.$)(?!.*\.\.)[a-zA-Z0-9_\-.]+$")
+ATTR_TYPE_REGEX = re.compile(r"^(String|Number|Binary)\..*$")
```

Comment:
Right, makes sense! Thanks a lot. I'll have a look at it and see what error AWS returns in that case, to maybe change the logic in the conditional, but I'll try to replicate! 👍

Reply:
Also I am nor sure if `.*` is correct here, since it should probably be `.+`. I would guess that e.g. `String.` is not allowed (but that is really only a guess, I don't know for sure).